sdks/java/nexmark/src/main/java/org/apache/beam/sdk/nexmark/NexmarkLauncher.java

```diff
@@ -747,6 +756,72 @@ public void processElement(ProcessContext c) {
       }));
   }
 
+  static final ParDo.SingleOutput<Event, byte[]> EVENT_TO_BYTEARRAY =
+          ParDo.of(new DoFn<Event, byte[]>() {
+            @ProcessElement
+            public void processElement(ProcessContext c) {
+              try {
+                byte[] encodedEvent = CoderUtils.encodeToByteArray(Event.CODER, c.element());
+                c.output(encodedEvent);
+              } catch (CoderException e1) {
+                LOG.error("Error while sending Event {} to Kafka: serialization error",
+                        c.element().toString());
+              }
+            }
+          });
+
+  /**
+   * Send {@code events} to Kafka.
+   */
+  private void sinkEventsToKafka(PCollection<Event> events) {
```

Comment:
This is currently unused, can you add something similar to Pubsub's PUBLISH_ONLY so users can generate and record the events in Kafka.